rust/otap-dataflow/crates/pdata/src/views/otlp/bytes/metrics.rs

```diff
@@ -214,7 +228,148 @@ impl FieldRanges for MetricFieldRanges {
     }
 }
 
-// TODO field ranges etc. for additional metrics messages
+/// Implementation of [`DataView`] backed by one of the buffers in Metric's "data" oneof field.
+pub struct RawData<'a> {
+    field_num: u64,
+    buf: &'a [u8],
+}
+
+/// Implementation of [`GaugeView`]
+pub struct RawGauge<'a> {
+    byte_parser: ProtoBytesParser<'a, GaugeFieldRanges>,
+}
+
+/// Known field ranges for fields on Gauge messages
+pub struct GaugeFieldRanges {
+    first_data_point: Cell<Option<(NonZeroUsize, NonZeroUsize)>>,
```

Comment:
Curious reader question: why are you using Cell<_> here and elsewhere in this file?